src/python/pants/help/help_info_extracter.py

```diff
@@ -478,10 +478,11 @@ def compute_choices(kwargs) -> tuple[str, ...] | None:
             return None
 
     @staticmethod
-    def get_first_provider(providers: tuple[str, ...] | None) -> str:
+    def get_provider(providers: tuple[str, ...] | None) -> str:
         if not providers:
             return ""
-        return providers[0]
+        # Pick the shortest backend name.
```

Comment:
Maybe link to the ticket where we discussed this and explain that if this algorithm isn't working well we can change to a better algorithm?

Reply:
I think that's what `git blame` is good for.. it'll link to this PR, which in turn has link to that discussion. wdyt?
Also, as John pointed out, perhaps there's a deeper issue we can resolve rather than making up better algorithms to work around this issue further..

Reply:
That's fine with me

Reply:
Sigh. It gets it wrong for one subsystem, at least: https://www.pantsbuild.org/v2.12/docs/reference-export-codegen

But that is a bit of an oddball I guess, as it doesn't have a natural home, but any backend that uses exportable codegen wants this subsystem registered.

Reply:
What if we changed it to be "Activated by any of these backends"? Maybe that's our issue -- the plugin API is intentionally very shareable. `export-codegen` is a good example! There is no one canonical backend for it.

Reply:
Good point, and we do have that information, it's just for the final stretch where I wanted to single out one responsible provider.